lib/msal-browser/test/interaction_client/SilentIframeClient.spec.ts

```diff
@@ -638,6 +639,124 @@ describe("SilentIframeClient", () => {
+                tenantId: testIdTokenClaims.tid || "",
+                scopes: TEST_CONFIG.DEFAULT_SCOPES,
+                idToken: testServerTokenResponse.id_token,
+                idTokenClaims: testIdTokenClaims,
+                accessToken: testServerTokenResponse.access_token,
+                fromCache: false,
+                correlationId: RANDOM_TEST_GUID,
+                expiresOn: new Date(
+                    Date.now() + testServerTokenResponse.expires_in * 1000
+                ),
+                account: testAccount,
+                tokenType: AuthenticationScheme.BEARER,
+            };
+            const testInvalidGrantError = new ServerError(
+                BrowserConstants.INVALID_GRANT_ERROR,
+                "First Server Error"
+            );
+            sinon
+                .stub(AuthorizationCodeClient.prototype, "getAuthCodeUrl")
+                .resolves(testNavUrl);
+            sinon
+                .stub(SilentHandler, "monitorIframeForHash")
+                .resolves(TEST_HASHES.TEST_SUCCESS_CODE_HASH_SILENT);
+            const handleCodeResponseStub = sinon
+                .stub(InteractionHandler.prototype, "handleCodeResponse")
+                .onFirstCall().rejects(testInvalidGrantError)
+                .onSecondCall().resolves(
+                    testTokenResponse
+                );
+            jest.spyOn(PkceGenerator, "generatePkceCodes").mockResolvedValue({
+                challenge: TEST_CONFIG.TEST_CHALLENGE,
+                verifier: TEST_CONFIG.TEST_VERIFIER,
+            });
+            jest.spyOn(BrowserCrypto, "createNewGuid").mockReturnValue(
+                RANDOM_TEST_GUID
+            );
+            const tokenResp = await silentIframeClient.acquireToken({
+                redirectUri: TEST_URIS.TEST_REDIR_URI,
+                loginHint: "testLoginHint",
+                prompt: PromptValue.NO_SESSION,
+            });
+            expect(tokenResp).toEqual(testTokenResponse);
+            expect(handleCodeResponseStub.calledTwice).toBeTruthy();
+            expect(handleCodeResponseStub.getCall(0).returnValue).toEqual(Promise.resolve(testInvalidGrantError));
+            expect(handleCodeResponseStub.getCall(1).args).toEqual(handleCodeResponseStub.getCall(0).args);
+        });
+
+        it("retries on invalid_grant error once and throws if still error", async () => {
+            const testInvalidGrantError = new ServerError(
+                BrowserConstants.INVALID_GRANT_ERROR,
+                "First Server Error"
+            );
+            sinon
+                .stub(AuthorizationCodeClient.prototype, "getAuthCodeUrl")
+                .resolves(testNavUrl);
+            sinon
+                .stub(SilentHandler, "monitorIframeForHash")
+                .resolves(TEST_HASHES.TEST_SUCCESS_CODE_HASH_SILENT);
+            const handleCodeResponseStub = sinon
+                .stub(InteractionHandler.prototype, "handleCodeResponse")
```

Comment:
instead of mocking this function, can we mock the post request? Will give us more coverage in our own code.

Reply:
Yes, thanks. Updated mocks.